blosc/blosc2.c

```diff
@@ -1333,7 +1333,7 @@ static void init_thread_context(struct thread_context* thread_context, blosc2_co
   thread_context->tmp = my_malloc(thread_context->tmp_nbytes);
   thread_context->tmp2 = thread_context->tmp + context->blocksize;
   thread_context->tmp3 = thread_context->tmp + context->blocksize + ebsize;
-  thread_context->tmp4 = thread_context->tmp + 2 * context->blocksize + ebsize;
+  thread_context->tmp4 = thread_context->tmp + (size_t)2 * context->blocksize + ebsize;
```

Comment:
Urgh.  This is a bit like back magic to me.  Could you shed some light on why this is needed?

Reply:
Need to cast 2 to a wider type `int32_t` -> `size_t`. If `int32_t` blocksize is sufficently large enough then it will overflow when multiplied by two.

Reply:
The same cast occurs on line 1332 right above it:
`thread_context->tmp_nbytes = (size_t)3 * context->blocksize + ebsize;`

Reply:
Ah, I think I got it now.  Thanks!